haystack/nodes/prompt/providers.py

```diff
@@ -239,12 +240,18 @@ def invoke(self, *args, **kwargs):
                     generated_texts[idx] = generated_texts[idx].replace(stop_word, "").strip()
         return generated_texts
 
-    def _ensure_token_limit(self, prompt: str) -> str:
+    def _ensure_token_limit(self, prompt: Union[str, List[Dict[str, str]]]) -> Union[str, List[Dict[str, str]]]:
         """Ensure that the length of the prompt and answer is within the max tokens limit of the model.
         If needed, truncate the prompt text so that it fits within the limit.
 
         :param prompt: Prompt text to be sent to the generative model.
         """
+        if isinstance(prompt, list) and len(prompt) > 0 and isinstance(prompt[0], dict):
```

Comment:
Did you experience this state? This should never happen because HFLocalInvocationLayer is never user when gpt turbo is used